tools/distpartitioning/dataset_utils.py

```diff
@@ -247,29 +383,53 @@ def get_dataset(input_dir, graph_name, rank, world_size, num_parts, schema_map):
     if dataset_features and (len(dataset_features) > 0):
         for etype_name, etype_feature_data in dataset_features.items():
             for feat_name, feat_data in etype_feature_data.items():
-                assert feat_data[constants.STR_FORMAT][constants.STR_NAME] in [constants.STR_NUMPY, constants.STR_PARQUET]
-                reader_fmt_meta = {"name": feat_data[constants.STR_FORMAT][constants.STR_NAME]}
-                num_chunks = len(feat_data[constants.STR_DATA])
-                read_list = np.array_split(np.arange(num_chunks), num_parts)
+                assert (feat_data[constants.STR_FORMAT][constants.STR_NAME]
+                    in [constants.STR_NUMPY, constants.STR_PARQUET])
+
+                edge_data = []
+                num_files = len(feat_data[constants.STR_DATA])
+                if num_files == 0:
+                    continue
+                read_list = np.array_split(np.arange(num_files), world_size)
```

Comment:
Same question here

Reply:
Not round-robin, but deterministic if `num_files` and `world_size` is fixed